taf/src/main/java/com/taf/automation/locking/SystemDateManager.java

```diff
@@ -0,0 +1,624 @@
+     * <OL>
+     * <LI>If weeks is less than one or existing ticket is not valid or
+     * plus days is less than equal to the existing ticket's days,
+     * then 0 is returned which indicates no reservation ticket</LI>
+     * <LI>If any actions are necessary to get the start date, then it is necessary to ensure
+     * they do not affect the system or the test</LI>
+     * <LI>This method should only be used if the test requires multiple clock moves
+     * as the initial state of the ticket will not block other clock moves</LI>
+     * <LI>The additional reservation will not be added until the reservation window closes</LI>
+     * </OL>
+     *
+     * @param existingTicket - Existing Reservation Ticket
+     * @param weeks          - Number of weeks to move clock forward in relation to the start date
+     * @param startDate      - The start date from which to add the weeks
+     * @return reservation ticket
+     */
+    public Long additionalReservationForWeeks(Long existingTicket, int weeks, Date startDate) {
+        Date endDate = DateUtils.addWeeks(startDate, weeks);
+        Long noOfDaysBetween = DateActions.daysBetween(startDate, endDate);
+        return additionalReservationForDays(existingTicket, noOfDaysBetween.intValue());
+    }
+
+    /**
+     * Open an additional reservation for plus specified months in relation to the start date of the test<BR>
+     * <B>Notes: </B>
+     * <OL>
+     * <LI>If months is less than one or existing ticket is not valid or
+     * plus days is less than equal to the existing ticket's days,
+     * then 0 is returned which indicates no reservation ticket</LI>
+     * <LI>If any actions are necessary to get the start date, then it is necessary to ensure
+     * they do not affect the system or the test</LI>
+     * <LI>This method should only be used if the test requires multiple clock moves
+     * as the initial state of the ticket will not block other clock moves</LI>
+     * <LI>The additional reservation will not be added until the reservation window closes</LI>
+     * </OL>
+     *
+     * @param existingTicket - Existing Reservation Ticket
+     * @param months         - Number of months to move clock forward in relation to the start date
+     * @param startDate      - The start date from which to add the months
+     * @return reservation ticket
+     */
+    public Long additionalReservationForMonths(Long existingTicket, int months, Date startDate) {
+        Date endDate = DateUtils.addMonths(startDate, months);
+        Long noOfDaysBetween = DateActions.daysBetween(startDate, endDate);
+        return additionalReservationForDays(existingTicket, noOfDaysBetween.intValue());
+    }
+
+    /**
+     * @return true if ticket reservations are being accepted else false
+     */
+    private boolean isOpenReservation() {
+        if (currentReservations >= MAX_RESERVATIONS) {
+            return false;
+        }
+
+        return clock.isNowBefore(getEndReservationTime());
```

Comment:
The end reservation time needs to be set when the reservation window is opened.  I found issues when running scenario 10 with 2 threads instead of 5.